### ISSUE TITLE
unable to connect to opds servers

### ISSUE THREAD

**Issue by tartanpion** [koreader/koreader#10504]
* KOReader version: last nightly
* Device: kobo libra2

#### Issue
can't search and download  from the gallica's opds 
#### Steps to reproduce
when trying to connect, freeze during a few minutes (look at the screenshot)

##### `crash.log` (if applicable)

[crash-opds.log](https://github.com/koreader/koreader/files/11594523/crash-opds.log)
![koreaderopds](https://github.com/koreader/koreader/assets/3924779/f3ccf78e-8e1e-4123-bfdd-4cfee2bc648d)

**Comment by NiLuJe** [comment]
It's been a while since I looked into any of this, but it appears to be returning 0 results (no matter the query, at a glance), while still offering a *next* page.

So, it looks like we honor that for a while, until it breaks for some reason (not that it would lead to actually actionable results).

TL;DR: Not sure if the server is up to specs there.

**Comment by NiLuJe** [comment]
(And since all the top-level links are implemented the same way, basically all of it is broken)

**Comment by hius07** [comment]
GALLICA E-BOOK CATALOG OPDS API
https://api.bnf.fr/fr/api-opds-du-catalogue-de-livres-numeriques-de-gallica

The example in the manual does not return the stream as described.

**Comment by NiLuJe** [comment]
Yeah, when I checked a few days ago, it returned no actual results, just the navigation/search skeleton, so we keep asking for more content and getting nothing actionable.

(And that was with outside requests via HTTPie or even firefox).

**Comment by NiLuJe** [comment]
At a very quick glance, the only thing we could do is handle the lack of `entry` tags in a saner manner so as not to loop indefinitely.

**Comment by hius07** [comment]
Works fine at the moment.
